monai/networks/nets/ahnet.py

```diff
@@ -0,0 +1,515 @@
+        pool3d_type: Type[nn.MaxPool3d] = Pool[Pool.MAX, 3]
+
+        self.conv3d_type = conv3d_type
+        self.norm3d_type = norm3d_type
+        self.relu_type = relu_type
+        self.pool3d_type = pool3d_type
+        # Make the 3x3x1 resnet layers
+        self.conv1 = conv3d_type(1, 64, kernel_size=(7, 7, 3), stride=(2, 2, 1), padding=(3, 3, 1), bias=False)
+        self.pool1 = pool3d_type(kernel_size=(1, 1, 2), stride=(1, 1, 2))
+        self.bn0 = norm3d_type(64)
+        self.relu = relu_type(inplace=True)
+        if deter_flag:
+            self.maxpool = pool3d_type(kernel_size=(2, 2, 2), stride=2)
+        else:
+            self.maxpool = pool3d_type(kernel_size=(3, 3, 3), stride=2, padding=1)
+
+        self.layer1 = self._make_layer(Bottleneck3x3x1, 64, layers[0], stride=1)
+        self.layer2 = self._make_layer(Bottleneck3x3x1, 128, layers[1], stride=2)
+        self.layer3 = self._make_layer(Bottleneck3x3x1, 256, layers[2], stride=2)
+        self.layer4 = self._make_layer(Bottleneck3x3x1, 512, layers[3], stride=2)
+
+        # Make the 3D dense decoder layers
+        densegrowth = 20
+        densebn = 4
+        ndenselayer = 3
+
+        num_init_features = 64
+        noutres1 = 256
+        noutres2 = 512
+        noutres3 = 1024
+        noutres4 = 2048
+
+        self.up0 = UpTransition(noutres4, noutres3, deter_flag)
+        self.dense0 = DenseBlock(
+            num_layers=ndenselayer, num_input_features=noutres3, bn_size=densebn, growth_rate=densegrowth, drop_rate=0.0
+        )
+        noutdense = noutres3 + ndenselayer * densegrowth
+
+        self.up1 = UpTransition(noutdense, noutres2, deter_flag)
+        self.dense1 = DenseBlock(
+            num_layers=ndenselayer, num_input_features=noutres2, bn_size=densebn, growth_rate=densegrowth, drop_rate=0.0
+        )
+        noutdense1 = noutres2 + ndenselayer * densegrowth
+
+        self.up2 = UpTransition(noutdense1, noutres1, deter_flag)
+        self.dense2 = DenseBlock(
+            num_layers=ndenselayer, num_input_features=noutres1, bn_size=densebn, growth_rate=densegrowth, drop_rate=0.0
+        )
+        noutdense2 = noutres1 + ndenselayer * densegrowth
+
+        self.trans1 = Projection(noutdense2, num_init_features)
+        self.dense3 = DenseBlock(
+            num_layers=ndenselayer,
+            num_input_features=num_init_features,
+            bn_size=densebn,
+            growth_rate=densegrowth,
+            drop_rate=0.0,
+        )
+        noutdense3 = num_init_features + densegrowth * ndenselayer
+
+        self.up3 = UpTransition(noutdense3, num_init_features, deter_flag)
+        self.dense4 = DenseBlock(
+            num_layers=ndenselayer,
+            num_input_features=num_init_features,
+            bn_size=densebn,
+            growth_rate=densegrowth,
+            drop_rate=0.0,
+        )
+        noutdense4 = num_init_features + densegrowth * ndenselayer
+
+        self.psp = PVP(noutdense4, deter_flag)
```

Comment:
looks like this is a PSP module https://arxiv.org/pdf/1612.01105.pdf, not sure why it is called PVP, need a reference for the PVP module if possible

Reply:
It comes from the author's official code: see https://github.com/lsqshr/AH-Net/blob/master/net3d.py (line 267), use PSP makes more sense I think, let me modify it.